File: icure-sdk/src/commonMain/kotlin/com/icure/sdk/model/TopicRole.kt
package com.icure.sdk.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
public enum class TopicRole() {
  @SerialName("PARTICIPANT")
  Participant,
  @SerialName("ADMIN")
  Admin,
  @SerialName("OWNER")
  Owner,
}

File: python-wrapper/src/python/model/TopicRole.py
from typing import Union, Dict
from enum import Enum

class TopicRole(Enum):
	Participant = "PARTICIPANT"
	Admin = "ADMIN"
	Owner = "OWNER"

	def __serialize__(self) -> object:
		return self.value

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'TopicRole':
		if data == "PARTICIPANT":
			return TopicRole.Participant
		elif data == "ADMIN":
			return TopicRole.Admin
		elif data == "OWNER":
			return TopicRole.Owner
		else:
			raise Exception(f"{data} is not a valid value for TopicRole enum.")

__all__ = ['TopicRole']